warehouse/malware/checks/package_turnover/check.py

```diff
@@ -0,0 +1,114 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from datetime import datetime, timedelta
+from textwrap import dedent
+
+from warehouse.accounts.models import UserEvent
+from warehouse.malware.checks.base import MalwareCheckBase
+from warehouse.malware.models import (
+    MalwareVerdict,
+    VerdictClassification,
+    VerdictConfidence,
+)
+from warehouse.packaging.models import ProjectEvent, Release
+
+
+class PackageTurnoverCheck(MalwareCheckBase):
+    version = 1
+    short_description = "A check for unusual changes in package ownership"
+    long_description = dedent(
+        """
+        This check looks at recently uploaded releases and determines
+        whether their owners have recently changed or decreased the security
+        of their accounts (e.g., by disabling 2FA).
+        """
+    )
+    check_type = "scheduled"
+    schedule = {"minute": 0, "hour": 0}
+
+    def __init__(self, db):
+        super().__init__(db)
+        self._scan_interval = datetime.utcnow() - timedelta(hours=24)
+
+    def user_posture_verdicts(self, project):
+        for user in project.users:
+            has_removed_2fa_method = self.db.query(
+                self.db.query(UserEvent)
+                .filter(UserEvent.user_id == user.id)
+                .filter(UserEvent.time >= self._scan_interval)
+                .filter(UserEvent.tag == "account:two_factor:method_removed")
+                .exists()
+            ).scalar()
+
+            if has_removed_2fa_method and not user.has_two_factor:
+                self.add_verdict(
+                    project_id=project.id,
+                    classification=VerdictClassification.Threat,
```

Comment:
I'm not sure disabling 2fa merits the `Threat` classification.

Reply:
I agree, although I'll point out that it's not just disabling 2FA -- it's disabling 2FA right before a new release and not adding a new method. But yes, I think it'd be a good idea to get a feel for the noise here and lower this if it's too much.

Reply:
I reviewed the last few days of activity for that tag, and was actually rather delighted to find that most instances of account:two_factor:method_removed were right before or after adding another method.

On average it's been fewer than ten instances per day.